pkg/manifests/types.go

```diff
@@ -104,7 +104,7 @@ type AlertmanagerMainConfig struct {
 }
 
 // The `K8sPrometheusAdapter` resource defines settings for the Prometheus Adapter component.
-// This is deprecated and will be removed in a future version.
+// This is deprecated config, setting this has no effect and will be removed in a future version.
```

Comment:
Keeping this to check in 4.18 release if any user defined config exists in cmo's configmap and then block upgrades till removed

Reply:
Can you add a TODO comment? It doesn't have to be 4.18 "exactly" but setting a target is good.